script.js

```diff
@@ -0,0 +1,42 @@
+var grid = document.querySelector("#grid");
+var whichPlayer = "X"
+var numMoves = 0
+grid.addEventListener("click", markSquare);
+
+
+
+function markSquare(square) {
+  if (square.target != square.currentTarget) {
+    var clicked = square.target.id;
+    playerMove(clicked);
+  }
+}
+
+function playerMove(clicked) {
+    if (document.getElementById(clicked).innerHTML !== ""){
+      alert("That square is already marked");
+    }
+    else if (whichPlayer === "O") {
+      //player 2 move
+      document.getElementById(clicked).innerHTML = "O";
+      document.getElementById(clicked).setAttribute("value", "O");
+      whichPlayer = "X";
+      checkWinner();
+    }
+    else {
+      //player 1 move
+      document.getElementById(clicked).innerHTML = "X";
+      whichPlayer = "O";
+      checkWinner();
+    }
+}
+
+function checkWinner() {
```

Comment:
i think you're on to something here for the `checkWinner()` function. I encourage you to flesh this out and see if you can get a working solution for the bonus!